### ISSUE TITLE
Issues about testset c-index, cox loss, and METABRIC dataset

### ISSUE THREAD

**Issue by centurion-crawler** [xhelenfu/DMGN_Survival_Prediction#1]
Hello Xiaohang,
Thanks for the great work and codes about survival analysis with IMC and PVs! I tried to reproduce the survival prediction results of single modal in your paper, but I found the following issues:

  first, I have a issue about the test set c-index calculation in the `test.py` code. Why does code choose the best score of test dataset？

`best_epoch_c_idx = np.argmax(c_idx_epochs_avg)` in `test.py`

  Second, does the cox loss calculation not use the survival time T in the patient's label of MetaData to calculate $j \in R(Ti)$ in $L_{surv}$? I tried to run the code and found that the loss would not decrease obviously.

  Third, where is the IMC tiff images in the [METABRIC dataset](https://idr.openmicroscopy.org/webclient/?show=project-1302) link to download? The export tiff button in the URL does not work(will refresh the webpage).

Looking forward to your reply！
Best,
Zhengfeng

**Comment by xhelenfu** [comment]
Hi Zhengfeng,

Thanks for your comments. 

1. The uploaded code finds which epoch produced the best performance for each cross-validation fold, which in our case is measured by the C-index.  

2. You are right, the loss calculation does not use the survival time as an input. More details may be found in Katzman et al. (2018) https://doi.org/10.1186/s12874-018-0482-1

3. The dataset may be downloaded from https://idr.openmicroscopy.org/ using accession code idr0076. Please see https://idr.openmicroscopy.org/about/download.html for more instructions. 

Best regards,
Xiaohang

**Comment by centurion-crawler** [comment]
Thanks for your reply! But I still have the problem:

Perhaps a reasonable validation approach should be to report the performance of cross-validation under a certain epoch, or to divide a certain proportion of samples from the training set again for checkpoint selection? Maybe choosing the best performance on test set lead to data leaks?

Best,
Zhengfeng

**Comment by xhelenfu** [comment]
Usually in k-fold cross-validation we do not select a certain epoch for different cross-validation folds since each fold uses different initial weights and training data. 

It is not necessary to divide the training set again if you're not tuning hyperparameters. If you wish to do tuning, you can replace the list of image filenames for individual folds (e.g., `1_test.txt` for fold 1) with the samples you want to use for tuning (make sure you change the training list too), then change back to the test samples for testing.

Best regards,
Xiaohang
